autoblocks/_impl/global_state.py

```diff
@@ -0,0 +1,46 @@
+import asyncio
+import threading
+from typing import Optional
+
+import httpx
+
+_client: Optional[httpx.AsyncClient] = None
+_loop: Optional[asyncio.AbstractEventLoop] = None
+_started: bool = False
+
+
+def init() -> None:
+    global _client, _loop, _started
+
+    if _started:
+        return
+
+    _client = httpx.AsyncClient()
+
+    _loop = asyncio.new_event_loop()
+
+    background_thread = threading.Thread(
+        target=run_event_loop,
+        args=(_loop,),
+        daemon=True,
+    )
+    background_thread.start()
+
+    _started = True
+
+
+def run_event_loop(_event_loop: asyncio.AbstractEventLoop) -> None:
```

Comment:
we should probably add an underscore here since it's not meant to be used outside of this module

Reply:
Updated